libfreerdp/core/capabilities.c

```diff
@@ -3806,12 +3805,13 @@ static BOOL rdp_read_capability_sets(wStream* s, rdpSettings* settings, UINT16 n
 			}
 		}
 
-		if (Stream_Pointer(s) != em)
+		rest = Stream_GetRemainingLength(&sub);
+		if (rest > 0)
 		{
 			WLog_ERR(TAG,
-			         "incorrect offset, type:0x%04" PRIX16 " actual:%" PRIuz " expected:%" PRIuz "",
-			         type, Stream_Pointer(s) - bm, em - bm);
-			Stream_SetPointer(s, em);
+			         "incorrect offset, type:0x%04" PRIX16 " actual:%" PRIuz " expected:%" PRIu16
+			         "",
+			         type, rest + length, length);
```

Comment:
probably that shoud have been `length - 4 - rest, length - 4`  but like above, just print the `rest` bytes not handled.

Reply:
No, `lenght` was already read and `rest` are the bytes still available.

Reply:
Sorry, correct, my fault.